velox/functions/sparksql/specialforms/SparkCastHooks.cpp

```diff
@@ -41,6 +40,12 @@ Expected<Timestamp> SparkCastHooks::castIntToTimestamp(int64_t seconds) const {
   return Timestamp(seconds, 0);
 }
 
+Expected<int64_t> SparkCastHooks::castTimestampToInt(
+    Timestamp timestamp) const {
+  return std::floor(
```

Comment:
Does timestamp.toMicros() / Timestamp::kMicrosecondsInSecond work?

Reply:
If the `timestamp.toMicros()` is negative, we need to round towards negative infinity instead of 0, so here we need `std::floor`, like the implementation in Spark using `Math.floorDiv(ts, MICROS_PER_SECOND)`